### PR TITLE
Add supported_features to the MQTT Alarm Control Panel

### PR DESCRIPTION
**Pull request by paravoid**
## Proposed change

This adds support for the new `supported_features` option of the MQTT Alarm Control Panel that I'm working on. https://github.com/home-assistant/core/pull/98363 has the full context of the change.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/98363
- Link to parent pull request in the Brands repository: N/A
- This PR fixes or closes issue: fixes N/A

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/alarm_control_panel.mqtt.markdown [hunk 1]
```diff
@@ -244,6 +244,11 @@ state_topic:
   description: The MQTT topic subscribed to receive state updates.
   required: true
   type: string
+supported_features
+  description: A list of features that the alarm control panel supports. The available options are "arm_home", "arm_away", "arm_night", "arm_vacation", "arm_custom_bypass", and "trigger".
+  required: true
```
Comment (jbouwh) [inline source/_integrations/alarm_control_panel.mqtt.markdown]:
```suggestion
  required: false
```

**Review by jbouwh — CHANGES_REQUESTED** [review]

File: source/_integrations/alarm_control_panel.mqtt.markdown [hunk 2]
```diff
@@ -244,6 +244,11 @@ state_topic:
...
+  type: [list, string]
```
Comment (jbouwh) [inline source/_integrations/alarm_control_panel.mqtt.markdown]:
```suggestion
  type: list
```

Unless you want to allow a single string. In that case you need to adjust the schema to ensure a list.

File: source/_integrations/alarm_control_panel.mqtt.markdown [hunk 3]
```diff
@@ -244,6 +244,11 @@ state_topic:
   description: The MQTT topic subscribed to receive state updates.
   required: true
   type: string
+supported_features
```
Comment (jbouwh) [inline source/_integrations/alarm_control_panel.mqtt.markdown]:
```suggestion
supported_features:
```

The missing `:` causes the lint errors

File: source/_integrations/alarm_control_panel.mqtt.markdown [hunk 4]
```diff
@@ -244,6 +244,11 @@ state_topic:
...
+  description: A list of features that the alarm control panel supports. The available options are "arm_home", "arm_away", "arm_night", "arm_vacation", "arm_custom_bypass", and "trigger".
```
Comment (jbouwh) [inline source/_integrations/alarm_control_panel.mqtt.markdown]:
```suggestion
  description: A list of features that the alarm control panel supports. The available list options are `arm_home`, `arm_away`, `arm_night`, `arm_vacation`, `arm_custom_bypass`, and `trigger`.
```

**Comment by jbouwh** [comment]
@paravoid  may be you can also adjust the example(s) and add the new option.

File: source/_integrations/alarm_control_panel.mqtt.markdown (hunk 2 above)
Comment (paravoid) [reply source/_integrations/alarm_control_panel.mqtt.markdown]: I erroneously thought `[list, string]` meant "a list of strings". Definitely not what I meant :) Thanks!

**Comment by paravoid** [comment]
> @paravoid may be you can also adjust the example(s) and add the new option.

Done

File: source/_integrations/alarm_control_panel.mqtt.markdown [hunk 5]
```diff
@@ -258,6 +263,24 @@ value_template:
...
+The example below shows a full configuration with an alarm panel that only supports the `arm_home` and `arm_away` features.
+
+{% raw %}
+
+```yaml
+# Example with partial feature support
+mqtt:
+  - alarm_control_panel:
+      name: "Alarm Panel"
+      supported_features: ["arm_home", "arm_away"]
```
Comment (jbouwh) [inline source/_integrations/alarm_control_panel.mqtt.markdown]:
```suggestion
      supported_features:
        - arm_home
        - arm_away
```

It is a YAML example

**Review by jbouwh — CHANGES_REQUESTED** [review]
Have a look at the suggestion for the example.

**Review by jbouwh — APPROVED** [review]
LGTM,
Can be merged as soon as the parent PR gets merged.
Thnx @paravoid 👍

**Pull request merged**